### ISSUE TITLE
--semVer level does not seem to be working

### ISSUE THREAD

**Issue by JustAnotherSoftwareDeveloper** [tjunnone/npm-check-updates#492]
- [X ] I have read the list of [known issues](https://github.com/tjunnone/npm-check-updates#known-issues) before filing this issue.
- [ X] I have searched for [similiar issues](https://github.com/tjunnone/npm-check-updates/issues) before filing this issue.

* node version: 10.13.0
* npm version: 6.8.0
* npm-check-updates version: 3.0.2

---------------------------

When I use the --semVer option, it states that all my dependencies are up to date. This is the same if I chose --semverLevel major or --semverLevel minor. When I just run ncu without specifying semverLevel, it notes a number of packages that can be upgraded.

**Comment by raineorshine** [comment]
Can you post the exact packages and versions? `--semverLevel` seems to be working fine for me. It is possible that there are no higher versions available within the major or minor version of your packages.  

```sh
raine[temp]$ cat package.json
{
  "dependencies": {
    "express": "2.4.1"
  }
}
raine[temp]$ ncu
Using /private/tmp/temp/package.json

 express  2.4.1  →  4.16.4

Run ncu -u to upgrade package.json
raine[temp]$ ncu --semverLevel major
Using /private/tmp/temp/package.json

 express  2.4.1  →  2.5.11

Run ncu -u to upgrade package.json
raine[temp]$ ncu --semverLevel minor
Using /private/tmp/temp/package.json

 express  2.4.1  →  2.4.7

Run ncu -u to upgrade package.json
```

**Comment by JustAnotherSoftwareDeveloper** [comment]
That's weird. I can duplicate the bug.
Here's my package.json:
```
{
  "name": "graph-view-web",
  "version": "0.0.1",
  "scripts": {
    "ng": "ng",
    "start": "ng serve --open --port 5500 --disable-host-check --public-host sdv-godzilla --host 0.0.0.0",
    "lint": "ng lint"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~7.0.0",
    "@angular/common": "~7.0.0",
    "@angular/compiler": "~7.0.0",
    "@angular/core": "~7.0.0",
    "@angular/forms": "~7.0.0",
    "@angular/http": "~7.0.0",
    "@angular/platform-browser": "~7.0.0",
    "@angular/platform-browser-dynamic": "~7.0.0",
    "@angular/router": "~7.0.0",
    "@mdi/font": "~3.5.95",
    "@ng-bootstrap/ng-bootstrap": "~4.0.4",
    "bandname": "~1.0.0",
    "bootstrap": "~4.3.1",
    "core-js": "^2.5.4",
    "corporate-ipsum": "~1.0.1",
    "deep-copy": "~1.4.2",
    "faker": "~4.1.0",
    "forcem-ipsum": "~1.7.3",
    "jeffsum": "0.0.1",
    "metal-name": "~1.3.1",
    "pirate-name-generator": "~1.0.0",
    "plausible-company-name": "~1.0.5",
    "rxjs": "~6.3.3",
    "samuel-ipsum": "~1.0.10",
    "zone.js": "~0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.10.0",
    "@angular/cli": "~7.0.4",
    "@angular/compiler-cli": "~7.0.0",
    "@angular/language-service": "~7.0.0",
    "@types/faker": "~4.1.5",
    "@types/node": "~8.9.4",
    "ajv": "~6.9.1",
    "ts-node": "~7.0.0",
    "tslint": "~5.11.0",
    "typescript": "~3.1.1"
  }
}
```
Running `ncu` returns:
```
 @angular/animations                 ~7.0.0  →     ~7.2.8 
 @angular/common                     ~7.0.0  →     ~7.2.8 
 @angular/compiler                   ~7.0.0  →     ~7.2.8 
 @angular/core                       ~7.0.0  →     ~7.2.8 
 @angular/forms                      ~7.0.0  →     ~7.2.8 
 @angular/http                       ~7.0.0  →     ~7.2.8 
 @angular/platform-browser           ~7.0.0  →     ~7.2.8 
 @angular/platform-browser-dynamic   ~7.0.0  →     ~7.2.8 
 @angular/router                     ~7.0.0  →     ~7.2.8 
 @ng-bootstrap/ng-bootstrap          ~4.0.4  →     ~4.1.0 
 core-js                             ^2.5.4  →     ^2.6.5 
 rxjs                                ~6.3.3  →     ~6.4.0 
 zone.js                            ~0.8.26  →    ~0.8.29 
 @angular-devkit/build-angular      ~0.10.0  →    ~0.13.5 
 @angular/cli                        ~7.0.4  →     ~7.3.5 
 @angular/compiler-cli               ~7.0.0  →     ~7.2.8 
 @angular/language-service           ~7.0.0  →     ~7.2.8 
 @types/node                         ~8.9.4  →   ~11.11.0 
 ajv                                 ~6.9.1  →    ~6.10.0 
 ts-node                             ~7.0.0  →     ~8.0.3 
 tslint                             ~5.11.0  →    ~5.13.1 
 typescript                          ~3.1.1  →  ~3.3.3333 
```

The semverLevel returns the same whether major or minor (with the obvious exception of the words major/minor): `All dependencies match the minor package versions :)`

npm-check-updates is now at 3.0.4. I updated right before I checked if this bug still existed.

**Comment by stoically** [comment]
The `--semverLevel` only supports [direct versions](https://github.com/tjunnone/npm-check-updates/blob/49e258a6c96b497aec6fb0796a54f6b69f81ff0d/lib/version-util.js#L154) like `1.2.3` and no ranges like `~1.2.3`. `findGreatestByLevel` would need to be refactored to use `semverutils.parseRange` instead to support ranges.

**Comment by JustAnotherSoftwareDeveloper** [comment]
Okay, my bad

**Comment by prma85** [comment]
Is there any plans to support version ranges? @raineorshine @stoically

**Comment by raineorshine** [comment]
@prma85 A PR would have to be initiated by a community member.

**Comment by raineorshine** [comment]
Added `--semverLevel` support for version ranges. Published in `v6.0.0`.
